### PR TITLE
Fixing build pruning tests

### PR DESCRIPTION
**Pull request by coreydaley**
Updating tests to look for 2-3 remaining builds based on HandleBuildPruning getting it's information from the cache.

### CODE REVIEW

File: test/extended/builds/build_pruning.go [hunk 1]
```diff
@@ -73,10 +73,15 @@ var _ = g.Describe("[builds][pruning] prune builds based on settings in the buil
 				fmt.Fprintf(g.GinkgoWriter, "%v", err)
 				return false, err
 			}
+			for _, build := range builds.Items {
+				fmt.Fprintf(g.GinkgoWriter, "%v: Status: %v, CompletionTimestamp: %v \n", build.Name, build.Status.Phase, *build.Status.CompletionTimestamp)
```
Comment (mfojtik) [inline test/extended/builds/build_pruning.go]: use the e2e.Logf() instead

**Comment by coreydaley** [comment]
@bparees ptal

File: test/extended/builds/build_pruning.go [hunk 2]
```diff
@@ -15,6 +15,9 @@ import (
 	exutil "github.com/openshift/origin/test/extended/util"
 )
 
+# These build pruning tests create 4 builds and check that 2-3 of them are left after pruning.
+# This variation in the number of builds that could be left is caused by the HandleBuildPruning
+# function using cached information from the buildLister.
```
Comment (bparees) [inline test/extended/builds/build_pruning.go]: // ?

**Comment by bparees** [comment]
fix the comment and lgtm

**Pull request merged**